flink-connectors/flink-connector-filesystem/src/main/java/org/apache/flink/streaming/connectors/fs/bucketing/BucketingSink.java

```diff
@@ -550,6 +550,9 @@ private void openNewPartFile(Path bucketPath, BucketState<T> bucketState) throws
 		Path inProgressPath = getInProgressPathFor(partPath);
 		if (bucketState.writer == null) {
 			bucketState.writer = writerTemplate.duplicate();
+			if (bucketState.writer == null) {
+				throw new RuntimeException("Could not duplicate writer.");
```

Comment:
I would add the class name of the `writerTemplate` object and that the class needs to implement the `Writer.duplicate()` method.